cmd2.py

```diff
@@ -438,6 +438,229 @@ def strip_ansi(text):
     return ANSI_ESCAPE_RE.sub('', text)
 
 
+def _pop_readline_history(clear_history=True):
+    """Returns a copy of readline's history and optionally clears it (default)"""
+    history = [
+        readline.get_history_item(i)
+        for i in range(1, 1 + readline.get_current_history_length())
+    ]
+    if clear_history:
+        readline.clear_history()
+    return history
+
+
+def _push_readline_history(history, clear_history=True):
+    """Restores readline's history and optionally clears it first (default)"""
+    if clear_history:
+        readline.clear_history()
+    for line in history:
+        readline.add_history(line)
+
+
+def _complete_from_cmd(cmd, text, line, begidx, endidx):
```

Comment:
cmd arg name shadows cmd module name from outer scope, best to be clear and rename argument